src/azure-cli/azure/cli/command_modules/network/_template_builder.py

```diff
@@ -34,13 +34,15 @@ def _build_frontend_ip_config(cmd, name, public_ip_id=None, subnet_id=None, priv
     return frontend_ip_config
 
 
-# pylint: disable=too-many-locals
+# pylint: disable=too-many-locals, too-many-statements
 def build_application_gateway_resource(cmd, name, location, tags, sku_name, sku_tier, capacity, servers, frontend_port,
-                                       private_ip_address, private_ip_allocation, cert_data, cert_password,
+                                       private_ip_address, private_ip_allocation,
```

Comment:
One question about the parameters for this function is that if we don't provide parameter value, what value the function will use. It can be seen as default value for parameters here.

Reply:
The default value would be `None`. The original custom function has the default value for all parameter. So the build function doesn't have default value.

Reply:
The default value would be None. The original custom function has the default value for all parameter. So the build function doesn't have default value.